src/main/java/org/wildfly/security/sasl/digest/DigestServerFactory.java

```diff
@@ -96,30 +96,30 @@ public SaslServer createSaslServer(String mechanism, String protocol, String ser
         }
 
         final String utf8 = (String)props.get(WildFlySasl.USE_UTF8);
-        Charset charset = (utf8 == null || Boolean.valueOf(utf8).booleanValue()) ? StandardCharsets.UTF_8 : StandardCharsets.ISO_8859_1;
+        Charset charset = (utf8 == null || Boolean.valueOf(utf8)) ? StandardCharsets.UTF_8 : StandardCharsets.ISO_8859_1;
```

Comment:
If you want to drop the ``booleanValue()`` then you should also change ``valueOf(utf8)`` to ``parseBoolean(utf8)`` to avoid useless boxing.

Reply:
Fixed.